controller_interface/include/controller_interface/helpers.hpp

```diff
@@ -48,7 +48,11 @@ bool get_ordered_interfaces(
     {
       if (!interface_type.empty())
       {
-        if ((name == interface.get_name()) && (interface_type == interface.get_interface_name()))
+        // check case where:
+        // (<joint> == <joint> AND <interface> == <interface>) OR <joint>/<interface> == 'full name'
+        if (
```

Comment:
We don't have any `else` clause here, wdyt about merging these conditions into `if (!interface_type.empty())` with an `&&`?

Reply:
This would be possible, but IMHO it would reduce readability and logic-following